### PR TITLE
Add documentation for version numbering and releases

### PR DESCRIPTION
**Pull request by Niederb**
I tried to come up with some documentation regarding the new versioning strategy and to make our release approach more explicit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -149,6 +149,21 @@ An example project on how to upgrade from older tags can be found in the Integri
...
+  - Note that this implies that there can be breaking changes in minor releases.
+- Patch/bugfix number is still used for bugfixes
+
+## Releases and backwards compatibility
+
+- We usually create a release for each new polkadot release
+  - This release contains all the features of our current `master` branch
+  - We don't create releases for polkadot bugfix releases unless there are known issues
+- We don't backport features and bugfixes per default. If requested we decide on a case by case basis.
+- New releases can contain breaking changes. We try to track these and announce them in the release notes.
```
Comment (masapr) [inline README.md:166]: The line
"We try to track these and announce them in the release notes"
I would change to 
"These are announced in the release notes."
Sure, in theory we might miss a breaking change. But generally I think we have a good procedure of tagging the PRs accordingly and therefore they also end up in our release notes.

File: README.md [hunk 2]
```diff
@@ -149,6 +149,21 @@ An example project on how to upgrade from older tags can be found in the Integri
 
 If you still experience issues during upgrading, do not hesitate to create an issue for support.
 
+## Version numbers
+
+We use a specififc versioning schema in order to enable us to release the same version multiple times for different polkadot releases
```
Comment (masapr) [inline README.md:154]: I would change this line to
Starting with release 1.16.0, we use a specific versioning schema...

**Review by masapr — CHANGES_REQUESTED** [review]

**Review by masapr — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Niederb) [reply README.md]: 👍 I changed it as you suggested

File: README.md (hunk 2 above)
Comment (Niederb) [reply README.md]: 👍 I changed it as you suggested

**Review by haerdib — COMMENTED** [review]
Could you also update the developer doc section?

Epsecially the https://github.com/scs/substrate-api-client/tree/master/docs#todo-list-for-a-new-release
and https://github.com/scs/substrate-api-client/tree/master/docs#semantic-versioning

or we can create an issue and do this in a separate PR

**Comment by Niederb** [comment]
@haerdib I updated the developer docs

**Review by haerdib — APPROVED** [review]
Cool, thanks a lot!

**Pull request merged**